### ISSUE TITLE
salt-cloud ec2 provider __opts__ is not defined

### ISSUE THREAD

**Issue by mlalpho** [saltstack/salt#28497]
provider:
```sls
ec2_west:
  driver: ec2
  id: xxx
  key: xxx
  private_key: key
  keyname: key
  ssh_username: ec2-user
  ssh_interface: private_ips
  location: us-west-2
  availability_zone: us-west-2b
  del_root_vol_on_destroy: True
  del_all_vols_on_destroy: True
  ssh_connect_timeout: 1800
```
profile:
```sls
ec2_dev:
  provider         : ec2_west
  image            : ami-amiid
  size             : t2.medium
  ssh_username     : ec2-user
  rename_on_destroy: True
  network_interfaces:
    - DeviceIndex: 0
      PrivateIpAddresses:
        - Primary: True
      SubnetId: subnet-xyz
      SecurityGroupId:
        - sg-12345
  block_device_mappings:
    - DeviceName    : /dev/sda1
      Ebs.VolumeSize: 100
      Ebs.VolumeType: gp2
      Ebs.SnapshotId: snap-12345
  minion:
    master       : xxxxx
    user         : root
    sudo_runas   : root
    sudo_password: ''
    grains:
      env: dev
```
Instance profile:
```sls
xxxxxxxxx:
  extends: ec2_dev
  tag: { 'Environment': 'Development', 'Group': 'test', 'App': 'test' }
```
```sh
[INFO    ] Finished executing state.highstate on the salt-minion
[INFO    ] Salt installed on xxxx
[INFO    ] Created Cloud VM 'xxxx'
[DEBUG   ]' xxxx' VM creation details:
{'amiLaunchIndex': '0',
 'architecture': 'x86_64',
 'blockDeviceMapping': None,
 'clientToken': None,
 'dnsName': None,
 'ebsOptimized': 'false',
 'groupSet': {'item': {'groupId': 'sg-gid',
                       'groupName': 'gname'}},
 'hypervisor': 'xen',
 'imageId': 'ami-amid',
 'instanceId': 'i-iid',
 'instanceState': {'code': '0', 'name': 'pending'},
 'instanceType': 't2.medium',
 'keyName': 'key',
 'launchTime': '2015-11-02T16:49:05.000Z',
 'monitoring': {'state': 'disabled'},
 'networkInterfaceSet': {'item': {'attachment': {'attachTime': '2015-11-02T16:49:05.000Z',
                                                 'attachmentId': 'xxx',
                                                 'deleteOnTermination': 'false',
                                                 'deviceIndex': '0',
                                                 'status': 'attaching'},
                                  'description': None,
                                  'groupSet': {'item': {'groupId': 'sg-gid',
                                                        'groupName': 'grp'}},
                                  'macAddress': 'xxxx',
                                  'networkInterfaceId': 'xxxx',
                                  'ownerId': 'xxxxx',
                                  'privateDnsName': 'xxxx',
                                  'privateIpAddress': 'xxxx',
                                  'privateIpAddressesSet': {'item': {'primary': 'true',
                                                                     'privateDnsName': 'xxxx',
                                                                     'privateIpAddress': 'xxxx'}},
                                  'sourceDestCheck': 'true',
                                  'status': 'in-use',
                                  'subnetId': 'xxx',
                                  'vpcId': 'vpc-xxxx'}},
 'placement': {'availabilityZone': 'us-west-2b',
               'groupName': None,
               'tenancy': 'default'},
 'privateDnsName': 'xxx',
 'privateIpAddress': 'xxxx',
 'productCodes': None,
 'reason': None,
 'rootDeviceName': '/dev/sda1',
 'rootDeviceType': 'ebs',
 'sourceDestCheck': 'true',
 'subnetId': 'xxxx',
 'virtualizationType': 'hvm',
 'vpcId': 'xxx'}
[DEBUG   ] Sending event - data = {'profile': 'cxxxx', '_stamp': '2015-11-02T16:56:08.550436', 'name': 'xxxx', 'instance_id': 'iid', 'provider': 'xxxx', 'event': 'created instance'}

[ERROR   ] There was a profile error: global name '__opts__' is not defined
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/cloud/cli.py", line 276, in run
    self.config.get('names')
  File "/usr/lib/python2.7/site-packages/salt/cloud/__init__.py", line 1403, in run_profile
    ret[name] = self.create(vm_)
  File "/usr/lib/python2.7/site-packages/salt/cloud/__init__.py", line 1260, in create
    conf_path = '/'.join(__opts__['conf_file'].split('/')[:-1])
NameError: global name '__opts__' is not defined
```
```
Salt Version:
           Salt: 2015.8.1

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Jun 24 2015, 00:41:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 7.1.1503 Core
        machine: x86_64
        release: 3.10.0-229.el7.x86_64
         system: CentOS Linux 7.1.1503 Core

```
`salt-cloud 2015.8.1 (Beryllium)`

related to #27956 ? or is this a different issue??

Am I doing something wrong? 

The instance creates successfully and highstate executes just fine. As of recently though, I'm see the opts not defined trace.

Please let me know if you need any more info.

Thank You.

**Comment by techhat** [comment]
Yup, looks like the same as #27956, and it has been patched.

**Comment by mlalpho** [comment]
@techhat How can I apply that patch? From what I can tell, it's been merged in to 2015.8.1, correct?

Do I need to reinstall?
`yum reinstall salt salt-cloud -y` didn't help any

My master's system:
```
cat /usr/lib/python2.7/site-packages/salt/cloud/__init__.py | grep __opts__
                conf_path = '/'.join(__opts__['conf_file'].split('/')[:-1])
```
`[ line 1260/2294 (54%), col 38/76 ...`

Am I missing something?

**Comment by techhat** [comment]
Well, it's in the 2015.8 branch, but i don't know if it's in 2015.8.1. @basepi, do you know? The PR in question is #27959.

If it's not in there, or you don't want to wait, just open up `salt/cloud/__init__` and change `__opts__` to `self.opts`. It's a tiny change.

**Comment by mlalpho** [comment]
aha! I misread the branch name. 

Okay, cool. I had a feeling I could just change the source file -- wanted to be 100% sure ! :)

Thanks guys/gals!

**Comment by rallytime** [comment]
Thanks @mlalpho! I'm going to go ahead and close this as a duplicate of #27956, and also because it is fixed by #27959. That fix will be available in the 2015.8.2 release.
